client/src/components/ConfigureWebhook/index.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  Chip,
+  CircularProgress,
+} from "@material-ui/core";
+import axios from "axios";
+import { rc4gitApiDomain } from "../../utils/constants";
+
+import "./index.css";
+
+export default function ConfigureWebhook(props) {
+  const [loading, setLoading] = useState(false);
+  const [selectedEvents, setSelectedEvents] = useState([
+    "push",
```

Comment:
@RonLek Are you by default creating the webhook? (When a channel gets created)
I think here, you must check if the webhook is already configured or not. If not, then this must be an empty array otherwise then you must fetch the webhook data from server and select the subscribed events.

Reply:
I have written the api to fetch webhook details, so you can use it from props.

Reply:
@djcruz93 No. The `selectedEvents` array is just the default set of subscription options that show up in the ConfigureWebhook dialog. The initialization here has nothing to do with what finally gets into the API call.
As for upgrading the webhook once it's created I'll be raising a PR for that soon where `selectedEvents` would hold those values.

Reply:
In my opionion, I think we must keep all unselected then by default as this will not confuse users because on updating webhook then also there will be pre selected hooks. Users can choose as they wish I don't think we must provide them with some preselection.

Reply:
I believe that's subjective. Btw even Gitter presents them pre-selected. We could maybe reduce the number of events to the most commonly used ones. What do you say to looking at user behavior and feedback over this one?

Reply:
Okay, then we can go ahead with the preselected for now. Actually, we are providing the update functionality of webhook and gitter doesn't provide it and our UI is the same as that of update and creation. So that's the reason I was saying to not provide pre-selection on creation as otherwise on updating the user will see the pre-selected which he selected on creation. 
You can ignore it, I think it won't make much difference until we get any feedback from actual users.